### PR TITLE
feat: online status handler

### PR DESCRIPTION
**Pull request by BiswaViraj**
### What change does this PR introduce?
Added a method to handle the online status of subscribers on connection and disconnection

### CODE REVIEW

File: apps/ws/src/socket/ws.gateway.ts [hunk 1]
```diff
@@ -39,4 +45,16 @@ export class WSGateway implements OnGatewayConnection {
...
+    await this.subscriberRepository.update(
+      { _id: subscriber._id, _organizationId: subscriber.organizationId },
+      {
+        $set: {
+          isOnline,
+          lastOnlineAt: new Date(),
+        },
+      }
+    );
+  }
```
Comment (scopsy) [inline apps/ws/src/socket/ws.gateway.ts:59]: Would suggest extracting this to a usecase or another dedicated service just so we don't start to build up the functionality in this file :)

File: apps/ws/src/socket/ws.gateway.ts [hunk 2]
```diff
@@ -28,7 +29,12 @@ export class WSGateway implements OnGatewayConnection {
       return this.disconnect(connection);
     }
 
-    return await connection.join(subscriber._id);
+    await connection.join(subscriber._id);
+    await this.handleOnlineStatus(subscriber, true);
+
+    connection.on('disconnect', async () => {
+      await this.handleOnlineStatus(subscriber, false);
+    });
```
Comment (LetItRock) [inline apps/ws/src/socket/ws.gateway.ts:37]: hey @BiswaViraj I'm missing here the logic that is described in the ticket, do you plan to add it in the separate PR?

Comment (BiswaViraj) [reply apps/ws/src/socket/ws.gateway.ts:37]: Had some issues with the multi-tab situation, initially intended to do it with a diff PR, will include it in this PR now.

File: apps/ws/src/shared/subscriber-online/subscriber-online.service.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import { SubscriberRepository } from '@novu/dal';
+
+@Injectable()
+export class SubscriberOnlineService {
+  constructor(private subscriberRepository: SubscriberRepository) {}
+
+  async handleConnection(subscriber: ISubscriberJwt) {
+    const isOnline = true;
+    const lastOnlineAt = new Date();
+    await this.updateOnlineStatus(subscriber, isOnline, lastOnlineAt);
```
Comment (LetItRock) [inline apps/ws/src/shared/subscriber-online/subscriber-online.service.ts:12]: we actually don't have to set the `lastOnlineAt` on the new connection, only when the disconnect happens

**Review by LetItRock — APPROVED** [review]

**Pull request merged**